### PR TITLE
Port nettype ansi

### PR DESCRIPTION
**Pull request by vasilissoti**
Based on SHA: fd57316160c9141eedc623272c0e8d3fdcbcf2e4 Adds SvNetType as an enum in structures as well as the function "port_nettype_ansi" for matching with the correct nettype found in the syntax tree (if any). If no explicit type is found then the LRM rules are followed. Finally nettype added to the display trait among the other port data.

### CODE REVIEW

File: src/structures.rs [hunk 1]
```diff
@@ -73,12 +73,31 @@ pub enum SvDataType {
...
+    Wor,
+    Wand,
+    Triand,
+    Trior,
+    Trireg,
+    Tri0,
+    Tri1,
+    Supply0,
+    Supply1,
+    NA,
```
[the remark this answers is not in the document]
Comment (vasilissoti) [reply src/structures.rs]: Just updated it.

**Comment by vasilissoti** [comment]
Branch merged with current state of upstream/master to eliminate any conflicts

File: src/main.rs [hunk 2]
```diff
@@ -401,6 +401,102 @@ fn port_datatype_ansi(
...
+    m: &sv_parser::AnsiPortDeclaration,
+    direction: &structures::SvPortDirection,
+    syntax_tree: &SyntaxTree,
+) -> Option<structures::SvNetType> {
+    let dir = unwrap_node!(m, AnsiPortDeclarationVariable, AnsiPortDeclarationNet);
+    match dir {
+        Some(RefNode::AnsiPortDeclarationVariable(_)) => return None, // "Var" token was found
+
+        Some(RefNode::AnsiPortDeclarationNet(x)) => {
+            let dir = unwrap_node!(x, NetType);
```
Comment (DaveMcEwan) [inline src/main.rs:414]: Is `dir` an appropriate name here? I don't think it refers to a direction.

File: .github/workflows/regression.yml [hunk 3]
```diff
@@ -5,7 +5,7 @@ jobs:
   build:
     strategy:
       matrix:
-        os: [ubuntu-latest, macOS-latest, windows-latest]
```
Comment (DaveMcEwan) [inline .github/workflows/regression.yml:8]: Can you make this a separate PR, please? Just to help keep track of the fact that Windows isn't tested in CI/CD.

**Pull request merged**